packages/compartment-mapper/test/test-bundle.js

```diff
@@ -75,3 +82,103 @@ test('equivalent archive behaves the same as bundle', async t => {
+    import.meta.url,
+  );
+  const sesShim = fs.readFileSync(sesShimLocation, 'utf8');
+  const bundle = await makeSecureBundle(read, fixture);
+  const log = [];
+  const print = entry => {
+    log.push(entry);
+  };
+  // bundle contains ses-shim and lockdown() call so we run in fresh Realm
+  const vmContext = vm.createContext({
+    print,
+    TextDecoder,
+    TextEncoder,
+  });
+  const vmEval = code => vm.runInContext(code, vmContext);
+  vmEval(sesShim);
+  vmEval('lockdown()');
+  const appExecPromise = vmEval(bundle);
+  const { namespace } = await appExecPromise;
+  t.deepEqual(namespace, {
+    xyz: 123,
+  });
+  t.deepEqual(log, expectedLog);
+});
+
+test('secure bundle from archive works', async t => {
+  const sesShimLocation = new URL(
+    '../../ses/dist/lockdown.umd.js',
+    import.meta.url,
+  );
+  const sesShim = fs.readFileSync(sesShimLocation, 'utf8');
+  const archiveBytes = await makeArchive(read, fixture);
+  const fakeArchiveLocation = new URL('app.agar', import.meta.url).toString();
+  const readWithArchive = async path => {
+    if (path === fakeArchiveLocation) {
+      return archiveBytes;
+    }
+    return read(path);
+  };
+  const readPowers = { read: readWithArchive };
+  const bundle = await makeSecureBundleFromArchive(
+    readPowers,
+    fakeArchiveLocation,
+  );
+  const log = [];
+  const print = entry => {
+    log.push(entry);
+  };
+  // bundle contains ses-shim and lockdown() call so we run in fresh Realm
+  const vmContext = vm.createContext({
+    print,
+    TextDecoder,
+    TextEncoder,
+  });
+  const vmEval = code => vm.runInContext(code, vmContext);
+  vmEval(sesShim);
+  vmEval('lockdown()');
+  const appExecPromise = vmEval(bundle);
+  const { namespace } = await appExecPromise;
+  t.deepEqual(namespace, {
+    xyz: 123,
+  });
+  t.deepEqual(log, expectedLog);
+});
+
+test('secure bundler safely sandboxes modules', async t => {
+  const sesShimLocation = new URL(
+    '../../ses/dist/lockdown.umd.js',
+    import.meta.url,
+  );
+  const appEntryLocation = new URL(
+    'fixtures-0/node_modules/bundle-unsafe/main.js',
+    import.meta.url,
+  );
+  const sesShim = fs.readFileSync(sesShimLocation, 'utf8');
+  const bundle = await makeSecureBundle(read, appEntryLocation);
+  // bundle contains ses-shim and lockdown() call so we run in fresh Realm
+  const vmContext = vm.createContext({
+    TextDecoder,
+    TextEncoder,
+  });
+  const vmEval = code => vm.runInContext(code, vmContext);
+  vmEval(sesShim);
+  vmEval('lockdown()');
+  const appExecPromise = vmEval(bundle);
+  const {
+    namespace: { myGlobalThis, myEval },
+  } = await appExecPromise;
+  // ensure the modules compartment global is not the realm global or context object
+  // nodejs vm implements a half-baked membrane with the vm realm globalThis and the context object
+  // wrapping in an array defeats the membrane
+  t.not(vmEval('globalThis'), myGlobalThis);
+  t.not(vmEval('[globalThis][0]'), myGlobalThis);
```

Comment:
i think that was a mistake
```suggestion
  t.not(vmEval('[globalThis]')[0], myGlobalThis);
```

Reply:
Or we could create a secure and insecure bundle and compare results.

Reply:
The first option sounds like the simplest way to get what we need?

Reply:
That requires 3 folders with package.json and index.js in each. That's why I thought of other suggestions to make the fixture smaller. It is though the closest thing to the actual subject of the test.

Reply:
opted to create an insecure bundle and compare behavior